automation/workflow/wftest/wftest.go

```diff
@@ -0,0 +1,263 @@
+				Name: "no_change",
+				Input: &dataset.Dataset{
+					Transform: &dataset.Transform{
+						Steps: []*dataset.TransformStep{
+							{
+								Name:     "transform",
+								Syntax:   "starlark",
+								Category: "transform",
+								Script:   "body = '''a,b,c\n1,2,3\n,4,5,6\n'''\ndef transform(ds,ctx):\n\tds.set_body(body,parse_as='csv')\n",
+							},
+						},
+					},
+				},
+			},
+		},
+		{
+			Name: "now",
+			Input: &workflow.Workflow{
+				ID:      "1d79b0ff-a133-4731-9892-5ee01842ca81",
+				OwnerID: "QmeL2mdVka1eahKENjehK6tBxkkpk5dNQ1qMcgWi7Hrb4B",
+				Created: &created,
+				Active:  true,
+			},
+			Expect: &workflow.Workflow{
+				ID:      "1d79b0ff-a133-4731-9892-5ee01842ca81",
+				OwnerID: "QmeL2mdVka1eahKENjehK6tBxkkpk5dNQ1qMcgWi7Hrb4B",
+				Created: &created,
+				Active:  true,
+			},
+			Dataset: &dstest.TestCase{
+				Name: "now",
+				Input: &dataset.Dataset{
+					Transform: &dataset.Transform{
+						Steps: []*dataset.TransformStep{
+							{
+								Syntax:   "starlark",
+								Category: "setup",
+								Name:     "setup",
+								Script:   "# load starlark dependencies\nload(\"encoding/csv.star\", \"csv\")\nload(\"time.star\", \"time\")\n",
+							},
+							{
+								Syntax:   "starlark",
+								Category: "transform",
+								Name:     "transform",
+								Script:   "# set the body\ndef transform(ds, ctx):\n  currentTime = time.now()\n  body = [\n    ['timestamp']\n  ]\n  body.append([str(currentTime)])\n  theCSV = csv.write_all(body)\n\n  ds.set_body(theCSV, parse_as='csv')\n",
+							},
+						},
+					},
+				},
+			},
+		},
+	}, nil
+}
+
+// LoadTestCases loads a directory of case directories
+func LoadTestCases(dir string) (tcs []*TestCase, err error) {
+	tcs = []*TestCase{}
+	fis, err := ioutil.ReadDir(dir)
+	if err != nil {
+		return
+	}
+	for _, fi := range fis {
+		if fi.IsDir() {
+			tc, err := NewTestCaseFromDir(filepath.Join(dir, fi.Name()))
+			if err != nil {
+				return nil, err
+			}
+			tcs = append(tcs, tc)
+		}
+	}
+	return
+}
+
+// NewTestCaseFromDir creates a test case from a directory of static test files
+// dir should be the path to the directory to check
+func NewTestCaseFromDir(dir string) (tc *TestCase, err error) {
+	dsTestCase, err := dstest.NewTestCaseFromDir(dir)
+	if err != nil {
+		return nil, err
+	}
+
+	tc = &TestCase{
+		Path:    dir,
+		Name:    filepath.Base(dir),
+		Dataset: &dsTestCase,
+	}
+
+	tc.Input, err = ReadWorkflow(dir, InputWorkflowFilename)
+	if err != nil {
+		return nil, fmt.Errorf("%s reading input workflow: %s", tc.Name, err)
+	}
+
+	tc.Expect, err = ReadWorkflow(dir, ExpectWorkflowFilename)
+	if err != nil {
+		if os.IsNotExist(err) {
+			err = nil
+		} else {
+			return nil, fmt.Errorf("%s: error loading expect workflow: %s", tc.Name, err)
+		}
+	}
+
+	preserve := TestCase{}
+	copier.Copy(&preserve, &tc)
+	testCaseCache = append(testCaseCache, tc)
+	return
+}
+
+// ReadWorkflow grabs a workflow for a given dir for a given filename
```

Comment:
I'd suggest changing the input parameter to be the full path to the file, and moving `filepath.Join` to the callers. It simplifies the behavior of this function in isolation, and greatly simplifies the comment describing this function.